core/src/main/java/io/questdb/cairo/AbstractDataFrameCursorFactory.java

```diff
@@ -47,12 +47,14 @@ public void toSink(CharSink sink) {
     }
 
     protected TableReader getReader(CairoSecurityContext sqlContext) {
-        return engine.getReader(
+        TableReader reader = engine.getReader(
```

Comment:
there are 139 instances where getReader() is used, the check should be done at pool level

Reply:
That would be too strict I believe and make failures in places where Reader is taken to compile a query or to read symbols in ILP.
This is the point where we are about to execute SELECT and checking capacity here will fail the read query